### PR TITLE
frege-repl 1.4-SNAPSHOT (new formula)

### PR DESCRIPTION
**Pull request by quephird**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [ ] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

Not quite; the target repo in question is just shy of the number of stars, forks, and watches required by this audit, at the time of this writing. After tweeting the Homebrew account Twitter, it seems that there is a little latitude given and so I am submitting the PR. However, I understand if this is rejected and I am quite happy to simply host a tap on my own GitHub account.

---

### CODE REVIEW

File: Formula/fregerepl.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    # Get rid of Windows batch file since Homebrew only runs on OS X.
+    rm_f Dir["bin/*.bat"]
+
+    # Move both bin and lib directories under libexec to avoid putting jars
+    # into /usr/local/lib (per Homebrew recommendations) while preserving
+    # the need for the shell script to reference jar files in a parallel
+    # directory and not having to tamper with the original shell script.
+    libexec.install "bin", "lib"
+
+    # Symlink to the real script.
```
Comment (mikemcquaid) [inline Formula/fregerepl.rb]: Can delete all these comments (and the blank lines in between) as they are self-explanatory, thanks!

File: Formula/fregerepl.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
+class Fregerepl < Formula
+  desc "Formula for installing the Frege REPL (not the compiler!!!)"
```
Comment (mikemcquaid) [inline Formula/fregerepl.rb]: Can remove the `!!!`

**Comment by DomT4** [comment]
I'd be fine on us accepting this re eligibility; last time I checked it was right on the cusp of acceptability and being a Java formula means minimal maintainer burden on our side.

**Comment by mikemcquaid** [comment]
> I'd be fine on us accepting this re eligibility; last time I checked it was right on the cusp of acceptability and being a Java(-related) formula means minimal maintainer burden on our side.

Agreed 👍

File: Formula/fregerepl.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
+class Fregerepl < Formula
```
Comment (UniqMartin) [inline Formula/fregerepl.rb]: The homepage and repository URL suggest the file should probably be named `frege-repl` instead, meaning the class name would also need to become `FregeRepl`. Any reason why you decided to omit the dash between `frege` and `repl`?

File: Formula/fregerepl.rb (hunk 2 above)
Comment (UniqMartin) [inline Formula/fregerepl.rb]: And you can also remove the “Formula for installing the“. That should be a description of the software being installed, not of the formula itself. I'd go as far as suggesting something completely different:

```ruby
desc "REPL (read-eval-print loop) for Frege"
```

I also think the “(not the compiler)” doesn't add any useful information. The description should be about what the software is and not about what it is not.

File: Formula/fregerepl.rb (hunk 3 above)
Comment (quephird) [inline Formula/fregerepl.rb]: I honestly didn't know what the naming conventions should have been as this is my first ever contribution to this project! I have changed everything per your suggestions. ^_^

File: Formula/fregerepl.rb (hunk 1 above)
Comment (quephird) [inline Formula/fregerepl.rb]: I suppose they were all for my own edification. All have been removed. ^_^

File: Formula/fregerepl.rb (hunk 3 above)
Comment (mikemcquaid) [inline Formula/fregerepl.rb]: Thanks!

File: Formula/fregerepl.rb (hunk 1 above)
Comment (mikemcquaid) [inline Formula/fregerepl.rb]: Cheers!

File: Formula/fregerepl.rb (hunk 2 above)
Comment (mikemcquaid) [inline Formula/fregerepl.rb]: (although we admire the enthusiasm 😉)

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format for new formulae is `frege-repl 1.4-SNAPSHOT (new formula)`. This format is detailed in the linked [Contributing guidelines](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md). Thanks!

**Comment by quephird** [comment]
OMG I'M SO EXCITED!!1!!!

**Pull request closed without merging**